In [ ]:
from google.colab import drive

In [ ]:
drive.mount('/content/drive/')

In [ ]:
from google.colab import drive
drive.mount('/content/drive')

In [ ]:
from tensorflow.keras.preprocessing.image import ImageDataGenerator

In [ ]:
train_datagen = ImageDataGenerator(rescale=1./255,zoom_range=0.2,horizontal_flip=True,vertical_flip=False,validation_split=0.2)

In [ ]:
test_datagen = ImageDataGenerator(rescale=1./255,validation_split=0.2)

In [ ]:
x_train=train_datagen.flow_from_directory(r"/content/drive/MyDrive/flowers",target_size=(64,64),class_mode='categorical',batch_size=100,subset = 'training')

In [ ]:
x_test=test_datagen.flow_from_directory(r"/content/drive/MyDrive/flowers",target_size=(64,64),class_mode='categorical',batch_size=100,subset = 'validation')

Found 444 images belonging to 5 classes.

In [ ]:
x_train.class_indices

{'daisy': 0, 'dandelion': 1, 'rose': 2, 'sunflower': 3, 'tulip': 4}

In [ ]:
from tensorflow.keras.models import Sequential

In [ ]:
from tensorflow.keras.layers import Dense,Convolution2D,MaxPooling2D,Flatten

In [ ]:
model=Sequential()

In [ ]:
model.add(Convolution2D(32,(3,3),input_shape=(64,64,3),activation='relu'))

In [ ]:
model.add(MaxPooling2D(pool_size=(2,2)))

In [ ]:
model.add(Flatten())

In [ ]:
model.summary()

Model: "sequential"
_________________________________________________________________
 Layer (type)                Output Shape              Param #   
 conv2d (Conv2D)             (None, 62, 62, 32)        896       
                                                                 
 max_pooling2d (MaxPooling2D  (None, 31, 31, 32)       0         
 )                                                               
                                                                 
 flatten (Flatten)           (None, 30752)             0         
                                                                 
Total params: 896
Trainable params: 896
Non-trainable params: 0
_________________________________________________________________


Model: "sequential"
_________________________________________________________________
 Layer (type)                Output Shape              Param #   
=================================================================
 conv2d (Conv2D)             (None, 62, 62, 32)        896       
                                                                 
 max_pooling2d (MaxPooling2D  (None, 31, 31, 32)       0         
 )                                                               
                                                                 
 flatten (Flatten)           (None, 30752)             0         
                                                                 
=================================================================
Total params: 896
Trainable params: 896
Non-trainable params: 0

In [ ]:
#hidden layers
model.add(Dense(300,activation='relu'))
model.add(Dense(150,activation='relu'))
model.add(Dense(75,activation='relu'))
model.add(Dense(5,activation='softmax'))#op layer

In [ ]:
model.compile(loss='categorical_crossentropy',optimizer='adam',metrics=['accuracy'])

In [ ]:
len(x_train)

18

In [ ]:
3457/100

34.57

In [ ]:
len(x_test)

5

In [ ]:
model.fit(x_train,steps_per_epoch=len(x_train),validation_data=x_test,validation_steps=len(x_test),epochs=30)

Epoch 1/30
18/18 [==============================] - 230s 13s/step - loss: 1.0800 - accuracy: 0.5162 - val_loss: 0.9077 - val_accuracy: 0.5113
Epoch 2/30
18/18 [==============================] - 20s 1s/step - loss: 0.7152 - accuracy: 0.6831 - val_loss: 0.7532 - val_accuracy: 0.6464
Epoch 3/30
18/18 [==============================] - 19s 1s/step - loss: 0.6492 - accuracy: 0.7083 - val_loss: 0.7347 - val_accuracy: 0.6757
Epoch 4/30
18/18 [==============================] - 18s 1s/step - loss: 0.6063 - accuracy: 0.7206 - val_loss: 0.8578 - val_accuracy: 0.6689
Epoch 5/30
18/18 [==============================] - 18s 1s/step - loss: 0.6686 - accuracy: 0.7010 - val_loss: 0.9111 - val_accuracy: 0.6441
Epoch 6/30
18/18 [==============================] - 19s 1s/step - loss: 0.5995 - accuracy: 0.7430 - val_loss: 0.7257 - val_accuracy: 0.6802
Epoch 7/30
18/18 [==============================] - 19s 1s/step - loss: 0.5714 - accuracy: 0.7520 - val_loss: 0.8746 - val_accuracy: 0.6577
Epoch 8/30
11/18 [=================>............] - ETA: 6s - loss: 0.5746 - accuracy: 0.7591

In [ ]:
model.save('flowers.h5')

In [ ]:
import numpy as np
from tensorflow.keras.models import load_model
from tensorflow.keras.preprocessing import image

In [ ]:
model=load_model('flowers.h5')

In [ ]:
img = image.load_img(r"/content/drive/MyDrive/flowers/sunflower/9904127656_f76a5a4811_m.jpg")

In [ ]:
img

In [ ]:
img = image.load_img(r"/content/drive/MyDrive/flowers/sunflower/9904127656_f76a5a4811_m.jpg",target_size(64,64))

In [ ]:
img

In [ ]:
x=image.img_to_array(img)

In [ ]:
x

array([[[ 94.,  61.,  30.],
        [137.,  98.,  57.],
        [141., 105.,  57.],
        ...,
        [181., 174., 119.],
        [186., 173., 120.],
        [185., 174., 118.]],

       [[130.,  81.,  38.],
        [155., 108.,  62.],
        [143., 105.,  56.],
        ...,
        [177., 171., 111.],
        [184., 174., 115.],
        [185., 168., 112.]],

       [[157., 107.,  56.],
        [152., 106.,  57.],
        [148., 107.,  53.],
        ...,
        [116.,  65.,  22.],
        [ 87.,  52.,   0.],
        [114.,  90.,  28.]],

       ...,

       [[245., 244., 249.],
        [244., 245., 249.],
        [218., 222., 225.],
        ...,
        [171., 142., 100.],
        [172., 144.,  94.],
        [165., 137.,  89.]],

       [[246., 248., 247.],
        [241., 242., 244.],
        [207., 207., 205.],
        ...,
        [173., 144., 100.],
        [176., 148., 100.],
        [171., 143.,  96.]],

       [[249., 249., 251.],
        [244., 248., 249.],
        [203., 191., 177.],
        ...,
        [164., 136.,  86.],
        [168., 140.,  93.],
        [169., 140.,  98.]]], dtype=float32)

In [ ]:
x.shape

(64, 64, 3)

In [ ]:
x=np.expand_dims(x,axis=0)
x.shape

(1, 64, 64, 3)

In [ ]:
y=np.argmax(model.predict(x),axis=1)

1/1 [==============================] - 12s 12s/step

In [ ]:
y


array([4])

In [ ]:
x_train.class_indices

{'daisy': 0, 'dandelion': 1, 'rose': 2, 'sunflower': 3, 'tulip': 4}

In [ ]:
img = image.load_img(r"/content/drive/MyDrive/flowers/daisy/5547758_eea9edfd54_n.jpg",target_size=(64,64))
print(img)
x=image.img_to_array(img)
print(x)
x=np.expand_dims(x,axis=0)
print(x.shape)
y=np.argmax(model.predict(x),axis=1)
print(x_train.class_indices)
print(y)